lib/active_model/serializer/adapter.rb

```diff
@@ -11,6 +12,47 @@ class Adapter
 
       attr_reader :serializer
 
+      # Only the Adapter class has these methods.
+      # None of the sublasses have them.
+      class << ActiveModel::Serializer::Adapter
+        def adapter_map
+          ADAPTER_MAP
+        end
+
+        def register(name, klass)
+          adapter_map.update(name.to_s.underscore => klass)
+        end
+
+        def get(adapter)
```

Comment:
> 
Cool, I was a little uncertain about the naming

So, the two methods havr slightly different behavior

The one in the adapter would return nil if none found, but the one in the serializer raises an argument error

So, i was thinking of having get and get! on the adapter

Needs some thought where nil is desired


>